Dixit_Logic/Classes/GameState.cs

```diff
@@ -0,0 +1,199 @@
+﻿using Dixit_Logic.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dixit_Logic.Classes
+{
+    /// <summary>
+    /// This class describes a state of game. Every property have internal set method
+    /// so only the logic can modify the game state.
+    /// </summary>
+    class GameState : IGameState
+    {
+        /// <summary>
+        /// Store the player who start the actual turn.
+        /// </summary>
+        private IPlayer _actPlayer;
+
+        /// <summary>
+        /// Store the deck which contains the cards what the players put on the table.
+        /// </summary>
+        private IDeck _boardDeck;
+
+        /// <summary>
+        /// Store a text what describes the association of turn's starter card.
+        /// </summary>
+        private string _cardAssocText;
+
+        /// <summary>
+        /// Store a dictionary associate a palyer with his/her guess.
+        /// </summary>
+        private Dictionary<IPlayer, ICard> _guesses;
+
+        /// <summary>
+        /// Store a dictionary associate a palyer with his/her own hand.
+        /// </summary>
+        private Dictionary<IPlayer, IDeck> _hands;
+
+        /// <summary>
+        ///  Store the deck from wich the players draw new cards.
+        /// </summary>
+        private IDeck _mainDeck;
+
+        /// <summary>
+        /// Store all players who playe in the actual game. 
+        /// </summary>
+        private IList<IPlayer> _players;
+
+        /// <summary>
+        /// Store a dictionary what associate a palyer with his/her own game-point value.
+        /// </summary>
+        private Dictionary<IPlayer, int> _points;
```

Comment:
Yes it could , i did it too. But Marci told me that it will be better in practice.